pkg/flowctl/data/filter.bpf.c

```diff
@@ -233,19 +234,47 @@ process_ipv4_tcp(struct __sk_buff *skb)
 }
 
 static inline int
-process_ipv4_icmp(struct __sk_buff *skb)
+process_ipv4_icmp(struct __sk_buff *skb, __u8 encap_with)
 {
   // bpf_printk("icmp packet");
   return TC_ACT_OK;
 }
 
 static inline int
-process_ipv4_udp(struct __sk_buff *skb)
+process_ipv4_udp(struct __sk_buff *skb, __u8 encap_with)
 {
   // bpf_printk("udp packet");
   return TC_ACT_OK;
 }
 
+static inline int
+process_ipv4_ipip(struct __sk_buff *skb)
+{
+  __u64 data = skb->data;
+  __u64 data_end = skb->data_end;
+  __u64 pkt_len = 0;
+
+  struct iphdr *outer_ih = (struct iphdr *)(data + sizeof(struct ethhdr));
+  assert_len(outer_ih, data_end);
+
+  struct iphdr *inner_ih = (struct iphdr *)((char *)outer_ih + sizeof(struct iphdr));
+  assert_len(inner_ih, data_end);
+
+  if (inner_ih->ihl < 5)
+    return TC_ACT_SHOT;
```

Comment:
If the IP header has options, `inner_ih->ihl` will be greater than 5, right? If you add the IP header (20) length in `process_ipv4_tcp`, you may need to implement it like the following?

```suggestion
  if (inner_ih->ihl != 5)
    return TC_ACT_SHOT;
```

Reply:
or you need to calculate correct offset with `ih->ihl`.

Reply:
Actually this tool supports IP option (ihl > 5) but current this PR doesn't calculate offset of inner IP with correct ihl. So let me fix it.